packages/frontend/app/(dashboard)/course/[cid]/insights/components/DashboardPresetComponent.tsx

```diff
@@ -0,0 +1,363 @@
+                matchingChart = gantt_charts[item.name]
+              }
+
+              const detailsNode: React.ReactNode = (
+                <div className={pillContainerClass}>
+                  <div className={pillLabelClass}>
+                    {(displayInfo.insightType == InsightType.Chart &&
+                      `${(matchingChart as ChartComponent)?.chartType} Chart`) ||
+                      (displayInfo.insightType == InsightType.Value &&
+                        'Single Value') ||
+                      (displayInfo.insightType == InsightType.Table &&
+                        'Table') ||
+                      (displayInfo.insightType == InsightType.GanttChart &&
+                        'Gantt Chart')}
+                  </div>
+                  <div className={pillLabelClass}>
+                    {((displayInfo.insightType == InsightType.Chart ||
+                      displayInfo.insightType == InsightType.GanttChart) &&
+                      `${
+                        matchingChart?.props.size
+                          ? translateSize(matchingChart.props.size)
+                          : 'Medium'
+                      }`) ||
+                      (displayInfo.insightType == InsightType.Value &&
+                        'Small') ||
+                      (displayInfo.insightType == InsightType.Table &&
+                        'Size Varies')}
+                  </div>
+                </div>
+              )
+              return (
+                <List.Item key={index} className="flex">
+                  <label className="flex flex-auto">
+                    <Checkbox
+                      checked={selectedInsights.includes(name)}
+                      onChange={() => toggleInsight(name)}
+                    />
+                    <div className="ml-2 flex flex-auto flex-row justify-between">
+                      <div className={'mr-4 flex flex-col gap-2'}>
+                        <div>
+                          <b>{name.replace(/([A-Z])/g, ' $1')}</b>
+                        </div>
+                        <div>{displayInfo.description}</div>
+                      </div>
+                      {detailsNode}
+                    </div>
+                  </label>
+                </List.Item>
+              )
+            }}
+            bordered
+          />
+        </Modal>
+      )}
+      <div className={'sticky top-2 z-10 float-right h-0 overflow-visible'}>
+        <div className="relative top-[-100%] flex flex-row">
+          <div className={isOpen ? isOpenClass : ''}>
+            <div className={'flex flex-row justify-between gap-3'}>
+              {isOpen && (
+                <span>
+                  <b>Dashboard Presets</b>
+                </span>
+              )}
+              <Button
+                className={
+                  'text-helpmeblue border-helpmeblue hover:bg-helpmeblue aspect-square h-min w-min border-0 p-1 hover:border-white hover:text-white'
```

Comment:
oh hey you found the helpmeblue custom tailwind color i made, seeing it being used makes me happy